PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,39 @@
+# PULL REQUEST TEMPLATE
+
+## Description
+
+Please provide a brief description of the changes made in this pull request.
+
+## Related Issue
+
+If this pull request is related to any GitHub issue, please mention it here with the appropriate issue number.
+
+## Proposed Changes
+
+List down the specific changes or features introduced in this pull request.
+
+- Feature/issue-number/feature-description: [Feature Description]
+- Refactor/issue-number/issue-description: [Issue Description]
+- ...
+
+## Additional Information
+
+If there's any additional information or context that the reviewers should know, please provide it here.
+
+## Checklist
+
+Please check the following before submitting your pull request:
+
+- [ ] I have tested my changes locally and they are working as expected.
+- [ ] I have added necessary documentation or updated existing documentation.
+- [ ] I have added relevant tests for the proposed changes.
+- [ ] My code follows the project's coding guidelines and style.
+- [ ] I have reviewed my own code and ensured its quality.
```

Comment:
Buradaki maddelere kodun formatlanması gerektiğini de ekliyor olmak gerekiyor, ilgili geliştirmeye ait unit testlerin yazılması gerektiğini de ekleyelim, sadece test yazılmış. Tüm bunlara ek olarak SonarLint ile birlikte kod içerisinde issue'lar bulunmamasına dikkat edilmesini ekleyelim. Var olan maddeleri buna uygun olacak şekilde güncelleyelim.

Reply:
Aklıma gelen diğer kısım da şu;

Pull Requestlerde;
- `Reviewers` zaten default olarak geliyor, boş olmamalı defaultlar olmalı.
- `Assignees` opsiyonel, olsa da olur olmasa da olur gibi.
- `Labels`, opsiyonel, olsa da olur olmasa da olur gibi.
- `Projects`, eğer bir issue açılmış ve onun için pull request açılmışsa olmamalı, fakat issue yoksa mutlaka BE Board'da olmalı ve sütun olarak `In Review` sütununda olmalı.
- `Milestone`, eğer bir issue açılmış ve onun için pull request açılmışsa olmamalı, fakat issue yoksa mutlaka doğru milestone tanımı için mutlaka pull request içerisinde belirtilmeli ve eklenmeli.

Reply:
I revised the Checklist part as shown below

## Checklist
Please check the following before submitting your pull request:

- [ ] I have tested my changes locally and they are working as expected.
- [ ] The code is formatted according to the project's coding guidelines and style.
- [ ] Necessary documentation has been added or existing documentation has been updated.
- [ ] Relevant unit tests have been written and included.
- [ ] Relevant integration tests have been written and included.
- [ ] The code has been reviewed to ensure its quality.
- [ ] The code does not contain any issues flagged by SonarLint.
- [ ] Default reviewers have been assigned to this pull request.
- [ ] Assignees have been added if necessary.
- [ ] Labels have been applied if necessary.
- [ ] If an issue is opened and this pull request is related to that issue, it should not be associated with any project board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project board (BE Board, In Review column) and milestone to track the progress.